### PR TITLE
SPUI-171 > Add market screen

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/app/screens/market/provider/gainers_market_items_pod.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../model/market_item_model.dart';
+import 'market_items_pod.dart';
+
+final gainersMarketItemsPod =
```
Comment (eli1stark) [inline lib/app/screens/market/provider/gainers_market_items_pod.dart:6]: Maybe call it `marketGainersPod`?

File: lib/app/screens/market/provider/loosers_market_items_pod.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../model/market_item_model.dart';
+import 'market_items_pod.dart';
+
+final loosersMarketItemsPod =
```
Comment (eli1stark) [inline lib/app/screens/market/provider/loosers_market_items_pod.dart:6]: Maybe call it `marketLosersPod`?

File: lib/app/screens/market/provider/loosers_market_items_pod.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Provider.autoDispose<List<MarketItemModel>>((ref) {
+  final marketReferences = ref.watch(marketItemsPod);
```
Comment (eli1stark) [inline lib/app/screens/market/provider/loosers_market_items_pod.dart:8]: Change `marketReferences` to `items`

File: lib/app/screens/market/provider/gainers_market_items_pod.dart [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    Provider.autoDispose<List<MarketItemModel>>((ref) {
+  final marketReferences = ref.watch(marketItemsPod);
```
Comment (eli1stark) [inline lib/app/screens/market/provider/gainers_market_items_pod.dart:8]: Change `marketReferences` to `items`

File: lib/app/screens/market/provider/gainers_market_items_pod.dart [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+  return marketReferences
+      .where((marketReference) => marketReference.dayPercentChange > 0)
```
Comment (eli1stark) [inline lib/app/screens/market/provider/gainers_market_items_pod.dart:11]: Change `marketReference` to `item`

File: lib/app/screens/market/provider/market_items_pod.dart [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../model/market_item_model.dart';
+import 'assets_spod.dart';
+import 'market_references_spod.dart';
+import 'prices_spod.dart';
+
+final marketItemsPod = Provider.autoDispose<List<MarketItemModel>>((ref) {
+  final marketReferences = ref.watch(marketReferencesSpod);
```
Comment (eli1stark) [inline lib/app/screens/market/provider/market_items_pod.dart:9]: call this variable just `references`

File: lib/app/screens/market/provider/market_items_pod.dart [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  final assets = ref.watch(assetsSpod);
+  final prices = ref.watch(pricesSpod);
+
+  final marketItems = <MarketItemModel>[];
```
Comment (eli1stark) [inline lib/app/screens/market/provider/market_items_pod.dart:13]: call just `items`

File: lib/app/screens/market/view/components/currency_button/components/currency_card.dart [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  const MarketItemCard({
+    Key? key,
+    required this.marketItem,
+  }) : super(key: key);
+
+  final MarketItemModel marketItem;
+
+  @override
+  Widget build(BuildContext context) {
+    final dayPercentageChange = marketItem.dayPercentChange > 0
```
Comment (eli1stark) [inline lib/app/screens/market/view/components/currency_button/components/currency_card.dart:17]: A good candidate for creating a seperate function

File: lib/app/screens/market/view/components/currency_button/components/currency_card.dart [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        horizontal: 14.w,
+      ),
+      margin: EdgeInsets.symmetric(vertical: 4.h),
+      decoration: BoxDecoration(
+          border: Border.all(
+            color: Colors.grey,
+          ),
+          borderRadius: const BorderRadius.all(
+            Radius.circular(20),
+          )),
```
Comment (eli1stark) [inline lib/app/screens/market/view/components/currency_button/components/currency_card.dart:33]: comma

File: lib/app/screens/market/view/components/header_text.dart [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class HeaderText extends StatelessWidget {
+  const HeaderText({Key? key, required this.text}) : super(key: key);
```
Comment (eli1stark) [inline lib/app/screens/market/view/components/header_text.dart:5]: comma, we use one-liner only if we have a single key and nothing else

File: lib/app/screens/market/view/components/market_tab.dart [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class MarketTab extends StatelessWidget {
+  const MarketTab({Key? key, required this.text}) : super(key: key);
```
Comment (eli1stark) [inline lib/app/screens/market/view/components/market_tab.dart:5]: comma

File: lib/app/screens/market/view/components/search_button.dart [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  Widget build(BuildContext context) {
+    return Ink(
+      decoration: ShapeDecoration(
+        color: Colors.grey.shade100,
+        shape: const CircleBorder(),
+      ),
+      child: IconButton(
+        icon: const Icon(Icons.search),
+        color: Colors.grey.shade600,
+        onPressed: () {},
```
Comment (eli1stark) [inline lib/app/screens/market/view/components/search_button.dart:16]: Will be done in future releases?

File: lib/app/screens/market/view/market.dart [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+const marketTabsLength = 4;
+
+class Market extends HookWidget {
+  const Market();
+
+  @override
+  Widget build(BuildContext context) {
+    final marketItems = useProvider(marketItemsPod);
+    final gainersMarketItems = useProvider(gainersMarketItemsPod);
+    final loosersMarketItems = useProvider(loosersMarketItemsPod);
```
Comment (eli1stark) [inline lib/app/screens/market/view/market.dart:23]: you can just simplify naming to items, gainers, losers.

File: lib/app/screens/market/view/market.dart [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+import '../provider/loosers_market_items_pod.dart';
+import '../provider/market_items_pod.dart';
+import 'components/currency_button/currency_button.dart';
+import 'components/header_text.dart';
+import 'components/market_tab.dart';
+import 'components/search_button.dart';
+
+const marketTabsLength = 4;
+
+class Market extends HookWidget {
```
Comment (eli1stark) [inline lib/app/screens/market/view/market.dart:16]: Refactor your carcass on top of TabBarView or refactor TabBarView itself. I prefer more the first variant.
So the idea is for us to abstract from UI details and see only necessary stuff like  ListView.builder for example.
AppBar, PreferedSized, and so on are just noise without explanation what this widget paints/represents.

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/screens/market/provider/gainers_market_items_pod.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/gainers_market_items_pod.dart:6]: done

File: lib/app/screens/market/provider/loosers_market_items_pod.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/loosers_market_items_pod.dart:6]: done

File: lib/app/screens/market/provider/loosers_market_items_pod.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/loosers_market_items_pod.dart:8]: done

File: lib/app/screens/market/provider/loosers_market_items_pod.dart [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+
+  return marketReferences
+      .where((marketReference) => marketReference.dayPercentChange < 0)
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/loosers_market_items_pod.dart:11]: done

File: lib/app/screens/market/provider/gainers_market_items_pod.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/gainers_market_items_pod.dart:8]: done

File: lib/app/screens/market/provider/gainers_market_items_pod.dart (hunk 5 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/gainers_market_items_pod.dart:11]: done

File: lib/app/screens/market/provider/market_items_pod.dart (hunk 6 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/market_items_pod.dart:9]: done

File: lib/app/screens/market/provider/market_items_pod.dart (hunk 7 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/provider/market_items_pod.dart:13]: done

File: lib/app/screens/market/view/components/currency_button/components/currency_card.dart (hunk 8 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/view/components/currency_button/components/currency_card.dart:17]: refactored

File: lib/app/screens/market/view/components/currency_button/components/currency_card.dart (hunk 9 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/view/components/currency_button/components/currency_card.dart:33]: done

File: lib/app/screens/market/view/components/header_text.dart (hunk 10 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/view/components/header_text.dart:5]: done

File: lib/app/screens/market/view/components/market_tab.dart (hunk 11 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/view/components/market_tab.dart:5]: done

File: lib/app/screens/market/view/components/search_button.dart (hunk 12 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/view/components/search_button.dart:16]: +

Comment (VovaMedvediev) [reply lib/app/screens/market/view/components/search_button.dart:16]: yes

File: lib/app/screens/market/view/market.dart (hunk 13 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/view/market.dart:23]: agree, refactored

File: lib/app/screens/market/view/market.dart (hunk 14 above)
Comment (VovaMedvediev) [reply lib/app/screens/market/view/market.dart:13]: refactored market_app_bar and market_tabs

**Review by eli1stark — APPROVED** [review]

**Pull request merged**